govcd/openapi_org_network_dhcp_binding.go

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2023 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// OpenApiOrgVdcNetworkDhcpBinding handles IPv4 and IPv6 DHCP bindings for NSX-T Org VDC networks.
+// Note. To create DHCP bindings, DHCP must be enabled on the network first (see
+// `OpenApiOrgVdcNetworkDhcp`)
+type OpenApiOrgVdcNetworkDhcpBinding struct {
+	OpenApiOrgVdcNetworkDhcpBinding *types.OpenApiOrgVdcNetworkDhcpBinding
+	client                          *Client
+	// ParentOrgVdcNetworkId is used to construct the URL for the DHCP binding as it contains Org
+	// VDC network ID in the path
+	ParentOrgVdcNetworkId string
+}
+
+// CreateOpenApiOrgVdcNetworkDhcpBinding allows to create DHCP binding for specific Org VDC network
+func (orgVdcNet *OpenApiOrgVdcNetwork) CreateOpenApiOrgVdcNetworkDhcpBinding(dhcpBindingConfig *types.OpenApiOrgVdcNetworkDhcpBinding) (*OpenApiOrgVdcNetworkDhcpBinding, error) {
+	client := orgVdcNet.client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointOrgVdcNetworksDhcpBindings
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, orgVdcNet.OpenApiOrgVdcNetwork.ID))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just a thought for the future: maybe we should think how to encapsulate this block into a single call, as it is repeating in different places. On the other hand, being explicit has some benefits too.

Reply:
I have some thoughts to experiment with generics for reusing a lot of it, but needs some experimentation. In some cases, we have to do some "custom" things (mainly handling some "one off" VCD API behaviors).